packages/eyes-sdk-core/lib/close/closeBatch.js

```
@@ -0,0 +1,33 @@
+'use strict'
+
+const ServerConnector = require('../server/ServerConnector')
+const Configuration = require('../config/Configuration')
+const Logger = require('../logging/Logger')
+const {presult} = require('../troubleshoot/utils')
+
+function setupServerConfig({serverUrl, apiKey, proxy}) {
+  const logger = new Logger(!!process.env.APPLITOOLS_SHOW_LOGS)
+  const configuration = new Configuration()
+  if (serverUrl) configuration.setServerUrl(serverUrl)
+  if (apiKey) configuration.setApiKey(apiKey)
+  if (proxy) configuration.setProxy(proxy)
+  return {configuration, logger}
+}
+
+async function closeBatch({
+  batchIds = [],
```

Not good. If user doesn't set batch ID's in the fluent API then you will silently not do anything.
You need to remove the default value and also have a test for this case (using the fluent API).
Notice the nuance in the test: the test for `BatchClose` which doesn't set batch ID's should throw an error that no batch ID's are set. So the `closeBatch` function that you inject needs to throw if there are not `batchIds`.

---

Sorry, it's my mistake about the test. It should only verify that `closeBatch` was called with `undefined`.